tests/spec/s-string.js

```diff
@@ -4,16 +4,18 @@ describe('String', function () {
     'use strict';
 
     describe('trim', function () {
-        var test = '\x09\x0A\x0B\x0C\x0D\x20\xA0\u1680\u180E\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200A\u202F\u205F\u3000\u2028\u2029\uFEFFHello, World!\x09\x0A\x0B\x0C\x0D\x20\xA0\u1680\u180E\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200A\u202F\u205F\u3000\u2028\u2029\uFEFF';
+        var wsp = '\t\n\v\f\r \xA0\u1680\u180E\u2000\u2001\u2002\u2003\u2004\u2005' +
+          '\u2006\u2007\u2008\u2009\u200A\u200B\u2028\u2029\u202F\u205F\u3000\uFEFF';
+        var test = [wsp, 'Hello, World!', wsp].join('');
 
         it('trims all ES5 whitespace', function () {
             expect(test.trim()).toEqual('Hello, World!');
             expect(test.trim().length).toBe(13);
         });
 
-        it('does not trim the zero-width space', function () {
-            expect('\u200b'.trim()).toBe('\u200b');
-            expect('\u200b'.trim().length).toBe(1);
+        it('does not trim the next-line character', function () {
```

Comment:
http://es5.github.io/#x7.2

> ECMAScript implementations must recognize all of the white space characters defined in Unicode 3.0. Later editions of the Unicode Standard may define other white space characters. ECMAScript implementations may recognize white space characters from later editions of the Unicode Standard.

I once read through several editions of the Unicode Standard and found that `U+0085` (NEL) has been in Cc ever since 1.1.5 at least, and in particular in 3.0 and all later versions, so although its purpose was a single-character replacement for `\r\n`, it's non-trimmable; also, as for actively trimming `U+200B` (ZWSP), that was in Zs before Unicode 4.0.1 (upon and after which it was in Cf), so ES5 must recognize it as trimmable (and ES6 must not): https://github.com/paulmillr/es6-shim/pull/354#issuecomment-115565060

(When I made that comment, I hadn't realized that even if an ES5 or ES6 implementation targets a later version of Unicode than the earliest permissible one, it must still recognize as whitespace all Zs characters from the earliest permissible version of Unicode, respectively 3.0 and 5.1.0.)